External/FEXCore/Source/Interface/Core/Core.cpp

```diff
@@ -1033,21 +1037,31 @@ namespace FEXCore::Context {
     }
   }
 
-  void FlushCodeRange(FEXCore::Core::InternalThreadState *Thread, uint64_t Start, uint64_t Length) {
+  static void InvalidateGuestThreadCodeRange(FEXCore::Core::InternalThreadState *Thread, uint64_t Start, uint64_t Length) {
+    std::lock_guard<std::recursive_mutex> lk(Thread->LookupCache->WriteLock);
+
+    auto lower = Thread->LookupCache->CodePages.lower_bound(Start >> 12);
+    auto upper = Thread->LookupCache->CodePages.upper_bound((Start + Length) >> 12);
```

Comment:
This is just copied from previous code, but shouldn't it be `(Start + Length - 1) >> 12`? `upper_bound` gets the first element *greater* than its argument, so it looks like the current code potentially iterates over one page too much.

Reply:
Quite possibly. This code path wasn't heavily used before so bugs might lurk.

Reply:
Updated for -1. Good catch there.